Mathlib/Tactic/Positivity/Basic.lean

```diff
@@ -415,3 +416,34 @@ def evalAscFactorial : PositivityExt where eval {u α} _ _ e := do
     assertInstancesCommute
     pure (.positive q(Nat.ascFactorial_pos $n $k))
   | _, _, _ => throwError "failed to match Nat.ascFactorial"
+
+open Rat
+
+private alias ⟨_, num_pos_of_pos⟩ := num_pos
+private alias ⟨_, num_nonneg_of_nonneg⟩ := num_nonneg
+private alias ⟨_, num_ne_zero_of_ne_zero⟩ := num_ne_zero
+
+/-- The `positivity` extension which identifies expressions of the form `Rat.num a`,
+such that `positivity` successfully recognises `a`. -/
+@[positivity Rat.num _]
+def evalRatNum : PositivityExt where eval {u α} _ _ e := do
```

Comment:
I thought we wanted to spread out the extensions over different files. I suppose the code ended up in this file because `Mathlib.Data.Rat.Order` can't import the required dependencies?

Reply:
Yes exactly. `Data.Rat.Order` will soon be imported by the definition of `Field` (because of `NNRat`), which is needed to give the `norm_num` support necessary for `positivity`. This makes it straight up impossible to put the extension in `Data.Rat.Order`, and means that the extra import is actually not so extra at all.